frontend/src/metabase/public/containers/PublicOrEmbeddedQuestion/PublicOrEmbeddedQuestionView.tsx

```diff
@@ -0,0 +1,119 @@
+import cx from "classnames";
+import { updateIn } from "icepick";
+import type { Dispatch, SetStateAction } from "react";
+
+import LoadingAndErrorWrapper from "metabase/components/LoadingAndErrorWrapper";
+import CS from "metabase/css/core/index.css";
+import { EmbedFrame } from "metabase/public/components/EmbedFrame";
+import type { DisplayTheme } from "metabase/public/lib/types";
+import QueryDownloadWidget from "metabase/query_builder/components/QueryDownloadWidget";
+import { PublicMode } from "metabase/visualizations/click-actions/modes/PublicMode";
+import Visualization from "metabase/visualizations/components/Visualization";
+import Question from "metabase-lib/v1/Question";
+import type Metadata from "metabase-lib/v1/metadata/Metadata";
+import type { UiParameter } from "metabase-lib/v1/parameters/types";
+import type {
+  Card,
+  VisualizationSettings,
+  Dataset,
+  ParameterId,
+  ParameterValuesMap,
+  DatasetQuery,
+} from "metabase-types/api";
+
+export function PublicOrEmbeddedQuestionView({
+  card,
+  metadata,
+  result,
+  uuid,
+  token,
+  getParameters,
+  parameterValues,
+  setParameterValue,
+  setParameterValueToDefault,
+  bordered,
+  hide_download_button,
+  hide_parameters,
+  theme,
+  titled,
+  setCard,
+}: {
+  initialized: boolean;
```

Comment:
After refactoring, some suggested types here were really weird, so it requires manual correction.